### ISSUE TITLE
Update CLI authentication to store login

### ISSUE THREAD

**Issue by codyaray** [StackStorm/st2#3110]
We use LDAP for authentication. I've been sharing this one-liner with all of our internal users:

```
export ST2_AUTH_TOKEN=$(st2 auth `whoami` -t)
```

As a new user this is definitely non-intuitive. It'd be really nice to support a mechanism like

```
st2 login
```

While you could require your username, you could also just default to the current user name since thats how many (most?) corporate environments would be setup with ldap anyway.

Or you can prompt with default like:

```
$ st2 login
Username [default <user>]: <prompt for input>
Password: <prompt for input>
```

Or skip straight to username and optional password

```
$ st2 login `whoami` -p $PASSWORD
```

Lots of ways to skin the cat. But the business objective here is to make it easier for new/occasional end-users to authenticate.

**Comment by Mierdin** [comment]
@codyaray It seems like much of what you suggested for `st2 login` is already supported by the existing `st2 auth` option - the only thing that's missing is that `st2 auth` doesn't export to `ST2_AUTH_TOKEN` by default.

If `st2 auth` had an option to auto-export this, would that solve this problem in your mind?

**Comment by codyaray** [comment]
Yep. Just didn't know if you wanted to piggyback or not. Either way works
for me. :)

-Cody

**Comment by Mierdin** [comment]
@codyaray The difficulty is that `st2client` won't be able to modify the environment of the parent shell due to the way environments work in Linux. This is why many tools that use this approach recommend you set the env yourself first.

We may be able to do something a little different to accomplish the same goal, however. Stay tuned here, I'll let you know if I come up with anything

**Comment by codyaray** [comment]
My thought would be writing the token to disk with a TTL using
filecachetools or something

**Comment by Mierdin** [comment]
@codyaray See #3123 - hopefully that is going in the right direction.

**Comment by codyaray** [comment]
awesome!

**Comment by Mierdin** [comment]
@codyaray Apologies for the delay on this. Lots of other things came up during the holiday season, but I wanted to make sure this got in before our next release. #3123 has been merged - I encourage you to check out the 2.2 release when it comes out and try the new feature.

Thanks for reporting this!

**Comment by codyaray** [comment]
Thank you!
